assets/src/components/mapPage/mapDisplay.tsx

```diff
@@ -0,0 +1,391 @@
+        <>
+          {showSelectionCard && (
+            <SelectionCardContainer>
+              <VehiclePropertiesCard
+                vehicleOrGhost={selectedVehicleOrGhost}
+                onClose={deleteSelection}
+              />
+            </SelectionCardContainer>
+          )}
+          {routePatternShape && <RouteShape shape={routePatternShape} />}
+          {isVehicle(selectedVehicleOrGhost) &&
+            (selectedVehicleOrGhost?.isShuttle ? (
+              <VehicleMarker
+                key={selectedVehicleOrGhost.id}
+                vehicle={selectedVehicleOrGhost}
+                isPrimary={true}
+                isSelected={true}
+              />
+            ) : (
+              <RouteVehicles
+                selectedVehicleRoute={selectedVehicleOrGhost.routeId}
+                selectedVehicleId={selectedVehicleOrGhost.id}
+                onPrimaryVehicleSelect={selectVehicle}
+              />
+            ))}
+        </>
+      )}
+
+      <InterruptibleFollower
+        onUpdate={onFollowerUpdate}
+        positions={position}
+        {...followerState}
+      />
+    </>
+  )
+}
+
+const SelectionDataLayers = ({
+  liveSelectedEntity,
+  showSelectionCard,
+  deleteSelection,
+  selectedRoutePattern,
+  setSelection,
+  setStateClasses,
+}: {
+  liveSelectedEntity: LiveSelectedEntity | null
+  showSelectionCard: boolean
+  selectedRoutePattern: RoutePattern | null
+  deleteSelection: () => void
+  setSelection: (selectedEntity: SelectedEntity | null) => void
+  setStateClasses: (classes: string | undefined) => void
+}) => {
+  switch (liveSelectedEntity?.type) {
+    case SelectedEntityType.Vehicle:
+      return (
+        <SelectedVehicleDataLayers
+          vehicleOrGhost={liveSelectedEntity.vehicleOrGhost}
+          showSelectionCard={showSelectionCard}
+          routePattern={selectedRoutePattern}
+          deleteSelection={deleteSelection}
+          selectVehicle={(vehicleOrGhost: VehicleOrGhost) =>
+            setSelection({
+              type: SelectedEntityType.Vehicle,
+              vehicleId: vehicleOrGhost.id,
+            })
+          }
+          setStateClasses={setStateClasses}
+        />
+      )
+    // TODO: handle SelectedEntityType.ROUTE
+    default:
+      return <MapElementsNoSelection setStateClasses={setStateClasses} />
+  }
+}
+
+const useSelectedEntityRoutePatterns = (
+  liveSelectedEntity: LiveSelectedEntity | null
+): [RoutePattern[] | null, RoutePattern | null] => {
+  const selectedEntityRouteData = selectedEntityRoutePattern(liveSelectedEntity)
+  const routePatterns: RoutePattern[] | null = useRoutePatterns(
+    selectedEntityRouteData?.routeId || null
+  )
+
+  const routePatternOfSelectedEntity =
+    (routePatterns &&
+      selectedEntityRouteData?.routePatternId &&
+      routePatterns?.find(
+        (rp) => rp.id === selectedEntityRouteData.routePatternId
+      )) ||
+    null
+
+  return [routePatterns, routePatternOfSelectedEntity]
```

Comment:
**question:** are there any places where we will actually use the full list of `routePatterns`?

Reply:
Closing the loop here from our sync discussion - these will be used in the upcoming RPC.